lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1404,7 +1404,26 @@ export class UserAgentApplication {
             }
             // if more than one cached token is found
             else if (filteredItems.length > 1) {
-                throw ClientAuthError.createMultipleMatchingTokensInCacheError(scopes.toString());
+                //find an exact match for domain
+                const requestDomain = UrlUtils.GetUrlComponents(serverAuthenticationRequest.authority).HostNameAndPort;
+                const filteredAuthorityItems: Array<AccessTokenCacheItem> = [];
+                for (let i = 0; i < filteredItems.length; i++) {
+                    const domain = UrlUtils.GetUrlComponents(filteredItems[i].key.authority).HostNameAndPort;
+                    if (domain === requestDomain) {
+                        filteredAuthorityItems.push(filteredItems[i]);
+                    }
+                }
```

Comment:
This can also be written using `filteredItems.filter`:

```js
const filteredAuthorityItems = filteredItems.find(filteredItem => {
  const domain = UrlUtils.GetUrlComponents(filteredItem.key.authority).HostNameAndPort;
  return domain === requestedDomain;
});

Reply:
for my own understanding, does using the find() function only return the first element where `domain === requestDomain`?

Reply:
Yes, sorry, meant to use `.filter` (.find is not supported in IE). Updated snippet.

Reply:
made the changes